src/table/components/PaginationContent.jsx

```diff
@@ -0,0 +1,189 @@
+
+import { handleLastTab } from '../utils/handle-key-press';
+
+const icons = {
+  FirstPage: FirstPageIcon,
+  PreviousPage: KeyboardArrowLeft,
+  NextPage: KeyboardArrowRight,
+  LastPage: LastPageIcon,
+  FirstPageRTL: LastPageIcon,
+  PreviousPageRTL: KeyboardArrowRight,
+  NextPageRTL: KeyboardArrowLeft,
+  LastPageRTL: FirstPageIcon,
+};
+
+export const shouldShow = (component, width) => {
+  switch (component) {
+    case 'selectPage':
+      return width > 700;
+    case 'rppOptions':
+      return width > 550;
+    case 'firstLast':
+      return width > 350;
+    case 'displayedRows':
+      return width > 250;
+    default:
+      return false;
+  }
+};
+
+export default function PaginationContent({
+  theme,
+  direction,
+  tableData,
+  pageInfo,
+  setPageInfo,
+  keyboard,
+  translator,
+  constraints,
+  footerContainer,
+  selectionsAPI,
+  rect,
+  handleChangePage,
+  lastPageIdx,
+  announce,
+}) {
+  const { totalRowCount, totalColumnCount, paginationNeeded } = tableData;
+  const { page, rowsPerPage, rowsPerPageOptions } = pageInfo;
+
+  if (!paginationNeeded) return null;
+
+  const onFirstPage = page === 0;
+  const onLastPage = page >= lastPageIdx;
+  const tabIndex = !keyboard.enabled || keyboard.active ? 0 : -1;
+  const width = footerContainer ? footerContainer.getBoundingClientRect().width : rect.width;
+  const showFirstAndLast = shouldShow('firstLast', width);
+  const showRowsPerPage = !selectionsAPI.isModal() && shouldShow('rppOptions', width) && totalColumnCount <= 100;
+  const displayedRowsText = translator.get('SNTable.Pagination.DisplayedRowsLabel', [
+    `${page * rowsPerPage + 1} - ${Math.min((page + 1) * rowsPerPage, totalRowCount)}`,
+    totalRowCount,
+  ]);
+
+  const handleChangeRowsPerPage = (evt) => {
+    setPageInfo({ ...pageInfo, page: 0, rowsPerPage: +evt.target.value });
+    announce({ keys: [['SNTable.Pagination.RowsPerPageChange', evt.target.value]], politeness: 'assertive' });
+  };
+
+  const handleSelectPage = (event) => handleChangePage(+event.target.value);
+
+  const handleLastButtonTab = keyboard.enabled ? (event) => handleLastTab(event, selectionsAPI.isModal()) : null;
+
+  const selectStyle = {
+    backgroundColor: 'inherit',
+    '& .MuiNativeSelect-icon': { color: theme.table.pagination.iconColor },
+  };
+
+  const getButton = (disabledCondition, pageNumber, type, onKeyDown = null) => {
+    const iconType = `${type}${direction === 'rtl' ? 'RTL' : ''}`;
+    const IconComponent = icons[iconType];
+    const buttonStyle = disabledCondition
+      ? {
+          color: theme.table.pagination.disabledIconColor,
+          cursor: 'default',
+        }
+      : { color: theme.table.pagination.iconColor };
+
+    return (
+      <IconButton
+        data-testid="pagination-action-icon-button"
+        onClick={!disabledCondition ? () => handleChangePage(pageNumber) : null}
+        aria-disabled={disabledCondition}
+        ok
```

Comment:
ok?

Reply:
I was getting a weird error about ok not being a valid prop, this is why 😅